nexus/src/external_api/http_entrypoints.rs

```diff
@@ -3727,6 +3729,78 @@ async fn instance_external_ip_list(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// Attach an external IP to an instance
+#[endpoint {
+    method = POST,
+    path = "/v1/instances/{instance}/external-ips/attach",
+    tags = ["instances"],
+}]
+async fn instance_external_ip_attach(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    path_params: Path<params::InstancePath>,
+    query_params: Query<params::OptionalProjectSelector>,
+    ip_to_detach: TypedBody<params::ExternalIpCreate>,
+) -> Result<HttpResponseAccepted<views::ExternalIp>, HttpError> {
+    let apictx = rqctx.context();
+    let handler = async {
+        let opctx = crate::context::op_context_for_external_api(&rqctx).await?;
+        let nexus = &apictx.nexus;
+        let path = path_params.into_inner();
+        let query = query_params.into_inner();
+        let instance_selector = params::InstanceSelector {
+            project: query.project,
+            instance: path.instance,
+        };
+        let instance_lookup =
+            nexus.instance_lookup(&opctx, instance_selector)?;
+        let ip = nexus
+            .instance_attach_external_ip(
+                &opctx,
+                &instance_lookup,
+                &ip_to_detach.into_inner(),
+            )
+            .await?;
+        Ok(HttpResponseAccepted(ip))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Detach an external IP from an instance
+#[endpoint {
+    method = POST,
+    path = "/v1/instances/{instance}/external-ips/detach",
+    tags = ["instances"],
+}]
+async fn instance_external_ip_detach(
+    rqctx: RequestContext<Arc<ServerContext>>,
+    path_params: Path<params::InstancePath>,
+    query_params: Query<params::OptionalProjectSelector>,
+    ip_to_detach: TypedBody<params::ExternalIpDelete>,
+) -> Result<HttpResponseAccepted<Option<views::ExternalIp>>, HttpError> {
```

Comment:
As mentioned [here](https://github.com/oxidecomputer/oxide.rs/pull/499#discussion_r1442168889), this should be something like

```rs
struct ExternalIpDetachment {
  external_ip: Option<ExternalIp>
}
```

Reply:
I agreed with @david-crespo that that's not the structure we probably want, however, I think the fix is to return an error response rather than a success response that indicates that there was no such entity. @david-crespo wrote up guidance for this: https://github.com/oxidecomputer/omicron/blob/main/docs/http-status-codes.adoc

Reply:
Agreed, I didn’t read far enough to see why it might be None

Reply:
That sounds reasonable enough for this case. My understanding is that the saga return value needs to remain `Ok(_)` due to idempotency requirements, but converting to another status code is simple just outside that.